tests/scalers/neo4j/neo4j_test.go

```diff
@@ -0,0 +1,296 @@
+	assert.NoErrorf(t, err, "cannot get cypher-shell version - %s", err)
+}
+
+func deployPodUp(t *testing.T, kc *kubernetes.Clientset) {
+	query := `CREATE (ac1:Person { name: "Danish", from: "Colombo", popularfor: "singer" }),
+	(ac2:Person { name: "Saanvi", from: "Delhi", popularfor: "actress" }),
+	(ac3:Person { name: "Saurav", from: "Mumbai", popularfor: "Badminton" }),
+	(ac4:Person { name: "Robert", from: "California", popularfor: "Swimming" }),
+	(ac5:Person { name: "Sezzi", from: "Florida", profession: "Doctor" }),
+	(ac6:Person { name: "Sally", from: "Texas", profession: "Software Engineer" }),
+	(ac7:Person { name: "Sohali", from: "Gandhinagar", popularfor: "Tiktok" }),
+	(ac8:Person { name: "Ranauk", from: "Bangalore", profession: "Entrepreneur" }),
+	(ac9:Person { name: "Soha", from: "Bhopal", profession: "Charter Accountant" }),
+	(ac10:Person { name: "Ananya", from: "Paris", popularfor: "Lawn Tennis" }),
+	(ac11:Person { name: "Anna", from: "Moscow", profession: "Software Engineer" }),
+	(ac1)-[:FOLLOWS]->(ac2),(ac2)<-[:FOLLOWS]-(ac1),(ac5)-[:FOLLOWS]->(ac1),
+	(ac6)-[:FOLLOWS]->(ac1),(ac8)-[:FOLLOWS]->(ac1),(ac9)-[:FOLLOWS]->(ac1),
+	(ac11)-[:FOLLOWS]->(ac1),(ac5)-[:FOLLOWS]->(ac2),(ac6)-[:FOLLOWS]->(ac2),
+	(ac5)-[:FOLLOWS]->(ac3),(ac6)-[:FOLLOWS]->(ac3),(ac8)-[:FOLLOWS]->(ac3),
+	(ac9)-[:FOLLOWS]->(ac3),(ac11)-[:FOLLOWS]->(ac3),(ac5)-[:FOLLOWS]->(ac4),
+	(ac6)-[:FOLLOWS]->(ac4),(ac8)-[:FOLLOWS]->(ac4),(ac9)-[:FOLLOWS]->(ac4),
+	(ac11)-[:FOLLOWS]->(ac4),(ac5)-[:FOLLOWS]->(ac7),(ac6)-[:FOLLOWS]->(ac7),
+	(ac5)-[:FOLLOWS]->(ac10),(ac6)-[:FOLLOWS]->(ac10),(ac8)-[:FOLLOWS]->(ac10),
+	(ac5)<-[:FOLLOWS]-(ac6),(ac5)<-[:FOLLOWS]-(ac8),(ac5)<-[:FOLLOWS]-(ac11),
+	(ac6)<-[:FOLLOWS]-(ac5),(ac6)<-[:FOLLOWS]-(ac8),(ac6)<-[:FOLLOWS]-(ac11),
+	(ac6)<-[:FOLLOWS]-(ac9),(ac11)<-[:FOLLOWS]-(ac9),(ac11)<-[:FOLLOWS]-(ac6),
+	(ac8)<-[:FOLLOWS]-(ac11),(ac8)<-[:FOLLOWS]-(ac9),(ac9)<-[:FOLLOWS]-(ac5),
+	(ac10)<-[:FOLLOWS]-(ac7),(ac4)<-[:FOLLOWS]-(ac7),(ac3)<-[:FOLLOWS]-(ac7),
+	(ac3)<-[:FOLLOWS]-(ac10),(ac1)<-[:FOLLOWS]-(ac7),(ac2)<-[:FOLLOWS]-(ac7),
+	(ac3)<-[:FOLLOWS]-(ac7),(ac3)<-[:FOLLOWS]-(ac4),(ac3)<-[:FOLLOWS]-(ac1),
+	(ac10)<-[:FOLLOWS]-(ac3),(ac10)<-[:FOLLOWS]-(ac2),(ac10)<-[:FOLLOWS]-(ac1)
+	return ac1,ac2,ac3,ac4,ac5,ac6,ac7,ac8,ac9,ac10,ac11`
+
+	podSpec := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "neo4j-demo-up",
+			Namespace: testNamespace,
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: "OnFailure",
+			Containers: []corev1.Container{
+				{
+					Name:  "neo4j-demo-up",
+					Image: "tanishabanik/neo4j-demo:0.0.6",
+					Args:  []string{fmt.Sprintf("neo4j://test-release.%s.svc.cluster.local:7687", testNamespace), query},
+				},
+			},
+		},
+	}
+
+	_, err := kc.CoreV1().Pods(testNamespace).Create(context.Background(), podSpec, metav1.CreateOptions{})
+	if err != nil {
+		assert.NoErrorf(t, err, "error in creating neo4j scale up pod: %s", err)
+	}
+}
```

Comment:
I have tried with jobs, it doesn't work somehow, I was using minikube and kind, both, same

Reply:
you can use a pod if you want, but use templates instead of kubeclient as we do in other tests please

Reply:
The only reason, I didn't use Job was because of templates, it somehow fails, though,  I will try it again

Reply:
be careful with the templates because (sadly) the indentation has to be with spaces and not tabs, otherwise it'll fail, maybe that's the problem. In the worst case, you can place a breakpoint in [the function](https://github.com/kedacore/keda/blob/main/tests/helper/helper.go#L434) and check the error to know why the template is failing

Reply:
Sure, I can try that

Reply:
worked with template